apache/incubator-zeppelin
# Leaflet support, based PR 152

### What is this PR for?
An update/follow-on for PR 152 that is limited to Leaflet support. There doesn't appear to have been any new movement on PR 152 since August 2015, so this PR updates Madhuka's work against master, but without any of the data validation or other non-Leaflet work.

### What type of PR is it?
Feature

### Todos
N/A

### Is there a relevant Jira issue?
ZEPPELIN-157

### How should this be tested?
Recreate Madhuka's example, e.g. use the following to populate a cell:
  https://gist.github.com/eddies/f20241e161aa2bbbb788

Then, in a new cell
```
%sql
select * from myMap
```
There should a new chart selector rightmost that will render the data w/ leaflet. You may need to clean your browser cache initially.

### Screenshots (if appropriate)
See Madhuka's original PR: https://github.com/apache/incubator-zeppelin/pull/152

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? Probably. It wasn't clear to me if PR 152 was going to move ahead in a new, Leaflet-only incarnation as suggested in the comment thread. But I needed Leaflet support, so went ahead with my own fork. I'm happy to work on documentation for this, but not sure if a sample notebook tutorial is what's desired (e.g. maybe something under the Display System docs is preferable?). If so, I'd just stick with something like the sample notebook in PR 152.
